### PR TITLE
[CWS] fix the example rule making use of `args_options`

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This rule is simply wrong:
- `~=` is not a valid operator, `=~` is
- the ticks are not ascii ticks making copying the rule not work out of the box
- you can't use `=~` on an array (you need to use `in`)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by modernplumbing — APPROVED** [review]

File: docs/cloud-workload-security/scripts/templates/agent_expressions.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ Examples:
 
 Examples:
 * `T=8` and `width=8` both are in *args_options* for the command `ls -T 8 --width=8`
-* `exec.args_options ~= [ “s=.*\’” ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
+* `exec.args_options in [ ~"s=.*\'" ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`
```
Comment (modernplumbing) [inline docs/cloud-workload-security/scripts/templates/agent_expressions.md:143]: wait doesn't this mean that the command ends in '? The \ is just to escape the '? Testing now

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md:143]: ah yeah, I think you found a third reason why this is wrong

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md:143]: this should most likely be `\\`

Comment (modernplumbing) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: i believe this is trying to detect this [vulnerability](https://www.sudo.ws/security/advisories/unescape_overflow/) ([additional resource](https://thesecmaster.com/fix-the-new-sudo-vulnerability-cve-2021-3156/)): `sudoedit -s /`. HOWEVER, the actual MITRE CVE [post](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3156) says "backslash" which is "\". In any case, my version of sudoedit doesn't even have -s as a valid flag anymore. Gonna ask the PDEs if they're familiar with this CVE and if it's supposed to be \ or /.

I'm trying with `exec.args_options in [ ~"e=.*\/" ]` with grep (idk other commands that have -s as a flag that can take args off the top of my head), and it's not working atm. will report back

Comment (modernplumbing) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: hmmm i'm not sure this works. I changed the functional test to 
```
{
			ID: "test_rule_args_options",
			//Expression: `exec.args_options in ["block-size=123"]`,
			Expression: `exec.args_options in [ ~"block-size=.*3" ]`,
},
```
at `pkg/security/tests/process_test.go`. and ran ` inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessContext/args-options"` which fails.

The test is executing `ls --block-size 123`

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: ah I know why, ~ is for pattern, and we are providing a regexp: `.*` is a regex

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: can you try with `r"block-size.."` ?

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: and `~"block-size=*3`

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: this should be good now in the last PR

Comment (modernplumbing) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: so i guess for this doc, we can do `exec.args_options in [ r"s=.*\/" ]`, assuming the vulnerability is indeed `sudoedit -s /` and MITRE is the one with the typo. `sudoedit -s \` I'm pretty sure would just get us into a new line unless sudoedit does some wild parsing of `\`

Comment (paulcacheux) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: I'm pretty sure the vuln is with a real backlash `\`. Reading https://blog.qualys.com/vulnerabilities-threat-research/2021/01/26/cve-2021-3156-heap-based-buffer-overflow-in-sudo-baron-samedit it seems:
- `\` is the underlying vulnerability
- `/` allows to check if your system is vulnerable ?

Comment (modernplumbing) [reply docs/cloud-workload-security/scripts/templates/agent_expressions.md]: ahhhh yes you're def right. consider my premature approval fully maturely approved

**Review by buraizu — APPROVED** [review]

**Pull request merged**